dx/usr/etc/zsh/zshrc

```diff
@@ -1,38 +1,214 @@
+#!/usr/bin/env zsh
+# vim:syntax=zsh
+# vim:filetype=zsh
+
 #
 # .zshrc is sourced in interactive shells.
 # It should contain commands to set up aliases,
 # functions, options, key bindings, etc.
 #
 
-if [[ -o interactive ]]; then
-  eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+# for profiling zsh
+# https://unix.stackexchange.com/a/329719/27109
+#
+# zmodload zsh/zprof
+
+# Load and initialize the completion system ignoring insecure directories with a
+# cache time of 20 hours, so it should almost always regenerate the first time a
+# shell is opened each day.
+autoload -Uz compinit
+_comp_path="${XDG_CACHE_HOME:-$HOME/.cache}/prezto/zcompdump"
+# #q expands globs in conditional expressions
+if [[ $_comp_path(#qNmh-20) ]]; then
+  # -C (skip function check) implies -i (skip security check).
+  compinit -C -d "$_comp_path"
+else
+  mkdir -p "$_comp_path:h"
+  compinit -i -d "$_comp_path"
+  # Keep $_comp_path younger than cache time even if it isn't regenerated.
+  touch "$_comp_path"
+fi
+unset _comp_path
+
+#
+# Options
+#
+
+setopt INC_APPEND_HISTORY # To save every command before it is executed
+setopt SHARE_HISTORY      # Share history between all sessions.
+setopt APPEND_HISTORY     # Append history to the history file.
+setopt COMPLETE_IN_WORD   # Complete from both ends of a word.
+setopt ALWAYS_TO_END      # Move cursor to the end of a completed word.
+setopt PATH_DIRS          # Perform path search even on command names with slashes.
+setopt AUTO_MENU          # Show completion menu on a successive tab press.
+setopt AUTO_LIST          # Automatically list choices on ambiguous completion.
+setopt AUTO_PARAM_SLASH   # If completed parameter is a directory, add a trailing slash.
+setopt EXTENDED_GLOB      # Needed for file modification glob modifiers with compinit.
+unsetopt MENU_COMPLETE    # Do not autoselect the first completion entry.
+unsetopt FLOW_CONTROL     # Disable start/stop characters in shell editor.
+
+#
+# Variables
+#
+
+export ZSHCONFIG=${ZDOTDIR:-$HOME}/.zsh
+
+ZSH_THEME="spaceship"
+
+# Standard style used by default for 'list-colors'
+LS_COLORS=${LS_COLORS:-'di=34:ln=35:so=32:pi=33:ex=31:bd=36;01:cd=33;01:su=31;40;07:sg=36;40;07:tw=32;40;07:ow=33;40;07:'}
+
+#
+# Styles
+#
+
+# zstyle ':omz:update' mode disabled  # disable automatic updates
+# zstyle ':omz:update' mode auto      # update automatically without asking
+zstyle ':omz:update' mode reminder # just remind me to update when it's time
```

Comment:
Do we use/want oh-my-zsh settings. I don’t _think_ they affect anything if it’s not installed but don’t think it’s needed.